src/components/signup/SignupForm.tsx

```diff
@@ -0,0 +1,204 @@
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirm] = useState("");
+  const [gender, setGender] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [registrationError, setRegistrationError] = useState(false);
+  const [registrationSuccess, setRegistrationSuccess] = useState(false);
+
+  const handleFirstNameChange = (e: FormEvent<HTMLInputElement>) => {
+    setFirstName(e.currentTarget.value);
+  };
+
+  const handleLastNameChange = (e: FormEvent<HTMLInputElement>) => {
+    setLastName(e.currentTarget.value);
+  };
+
+  const handleUsernameChange = (e: FormEvent<HTMLInputElement>) => {
+    setUsername(e.currentTarget.value);
+  };
+
+  const handleEmailChange = (e: FormEvent<HTMLInputElement>) => {
+    setEmail(e.currentTarget.value);
+  };
+
+  const handlePasswordChange = (e: FormEvent<HTMLInputElement>) => {
+    setPassword(e.currentTarget.value);
+  };
+
+  const handleConfirmPasswordChange = (e: FormEvent<HTMLInputElement>) => {
+    setConfirm(e.currentTarget.value);
+  };
+
+  const handleGenderChange = (e: FormEvent<HTMLSelectElement>) => {
+    setGender(e.currentTarget.value);
+  };
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+
+    if (
+      firstName === "" ||
+      lastName === "" ||
+      username === "" ||
+      email === "" ||
+      password === "" ||
+      gender === "" ||
+      confirmPassword != password
+    ) {
+      setError(true);
+      return;
+    } else {
+      setError(false);
+      setIsLoading(true);
+
+      const userData = {
+        user: {
+          first_name: firstName,
+          last_name: lastName,
+          username,
+          email,
+          password,
+          password_confirmation: confirmPassword,
+          gender,
+        },
+      };
+
+      try {
+        const response = await registerUser(userData);
+        console.log(response);
+        setRegistrationSuccess(true);
+        // Redirect to confirmation sent page
+      } catch (error) {
+        console.error(error);
+        setRegistrationError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+  };
+
+  return (
+    <form>
+      <div className="flex w-[180px] max-w-full flex-col mt-20 self-end max-md:mt-10 mb-8">
+        <div className="text-black text-3xl font-bold tracking-[3px] self-center">
+          SIGN UP
+        </div>
+        <div className="text-black text-center text-xs font-semibold tracking-widest self-center mt-5">
+          <label htmlFor="firstName"> FIRST NAME </label>
+          <input
+            id="firstName"
+            type="text"
+            className="self-stretch flex shrink-0 h-[37px] flex-col mt-2 border-[0.5px] border-solid border-zinc-950 p-3"
+            aria-label="Name"
+            onChange={handleFirstNameChange}
+          />
+        </div>
```

Comment:
Maybe creating a common input component will be a good idea here, as you are repeating most of the code